internal/manifests/collector/ports.go

```diff
@@ -184,8 +174,23 @@ func getReceiverServicePort(logger logr.Logger, serviceAddress string, receiverN
 		if err != nil {
 			logger.Error(err, "error parsing port from endpoint for receiver", zap.String("endpoint", serviceAddress), zap.String("receiver", receiverName))
 		} else {
-			if _, ok := servicePortsMap[port]; ok {
-				logger.Info("Duplicate port has been configured in Agent Config for port", zap.Int32("port", port))
+			if ports, exists := servicePortsMap[port]; exists {
+				for _, existingPort := range ports {
+					if existingPort.Protocol == protocol {
+						logger.Info("Duplicate port and protocol combination configured", zap.Int32("port", port), zap.String("protocol", string(protocol)))
+						return
+					}
+				}
+				name := CWA + receiverName
+				if receiverName == OtlpGrpc || receiverName == OtlpHttp {
+					name = fmt.Sprintf("%s-%d", receiverName, port)
+				}
+				sp := corev1.ServicePort{
+					Name:     name,
+					Port:     port,
+					Protocol: protocol,
+				}
+				servicePortsMap[port] = append(servicePortsMap[port], sp)
```

Comment:
nit: The port assignment doesn't need to be duplicated here. The check will already skip the append if it's a duplicate port/protocol. Same thing with the changes below.

```suggestion
			if ports, exists := servicePortsMap[port]; exists {
				for _, existingPort := range ports {
					if existingPort.Protocol == protocol {
						logger.Info("Duplicate port and protocol combination configured", zap.Int32("port", port), zap.String("protocol", string(protocol)))
						return
					}
				}
			}
			name := CWA + receiverName
			if receiverName == OtlpGrpc || receiverName == OtlpHttp {
				name = fmt.Sprintf("%s-%d", receiverName, port)
			}
			sp := corev1.ServicePort{
				Name:     name,
				Port:     port,
				Protocol: protocol,
			}
			servicePortsMap[port] = append(servicePortsMap[port], sp)
```